openedx/frontend-app-publisher
# feat: Add Product Source as a read only field in publisher

[PROD-3106](https://2u-internal.atlassian.net/browse/PROD-3106)

This PR adds a default product source `edX` to every course created through the publisher. We are hiding the field `Product Source` in the create course form for the time being.
In addition, this PR adds a non-editable field named `Source` to the course edit form. This field is displayed only if the course has a source else `N/A`. The source is displayed on the course edit page.


Testing Instructions
--------------------

1. Pull the changes https://github.com/openedx/course-discovery/pull/3766 locally and run the discovery service.
  - `make dev.up.discovery`
2. Pull the changes of the current PR locally and run the frontend-app-publisher and studio.
   - `make dev.up.frontend-app-publisher`
   - `make dev.up.studio`

3. Create a new course in the publisher app.
4. Verify that the `Source` field is displayed in the course edit form and is non-editable.
5. Verify that the `Source` field is displayed in the course edit page.

<img width="848" alt="image" src="https://user-images.githubusercontent.com/78806673/227935693-93754465-fc59-4b4b-aa1e-65511e9596f0.png">

<img width="1107" alt="image" src="https://user-images.githubusercontent.com/78806673/224568834-a9ec33a8-2cb9-492a-a982-f8259bfecaab.png">

-----
Vedio
https://user-images.githubusercontent.com/78806673/227967220-55165a09-6344-40b5-b59c-b14fb1c0a275.mov
